### DISCUSSION TITLE
Update module on related module creation

### DISCUSSION THREAD

**Discussion by rochmirabeau** [YetiForceCompany/YetiForceCRM#15510] [How to] [answered]
I have two modules, A and B:
Module A has a one-to-many relation to module B. 
Module B has a custom function that executes on creation to calculate values.
Module A has a custom function to fetch and sum totals from all related Module B records

When I Add Record of module b from module a, the totals are not updated; however, when I edit module B or module A, the totals are updated.

How can I make the totals of module A update when creating a related module B from the add record menu?

**Comment by vovpff** [comment, marked as answer]
What you mean for custom function?

**Reply by rochmirabeau** [reply]
like the following (edit)

    class ModuleBWorkFlow extends VTTask {
        public function updateModuleA($recordModel) {
            $relatedModuleName = "ModuleA";
            $relatedModuleBs = (new \App\Db\Query())->select(['crmid'])->from('vtiger_crmentityrel')->where(['relcrmid' => $recordModel->get("id")])->createCommand()->query();
            while ($row = $relatedWorkOrders->read()) {
                $id = (string) implode(" ", $row);  
                $update = Vtiger_Record_Model::getInstanceById($id, $relatedModuleName);
                $update->set('testfield', "Module B told me to update");
            }   
        }   
    }

**Reply by vovpff** [reply]
You can try to use handlers instead workflow tasks. Check /modules/modulename/handlers for examples. Contacts and payments in modules should have handlers.

For me handlers more easy and flexible.
